common/p_enemy.cpp

```diff
@@ -2336,26 +2349,36 @@ void A_Face(AActor* mo)
 
 void A_Scratch(AActor* mo)
 {
-	/* [AM] Not implemented...yet.
-	mo->target && (A_FaceTarget(mo), P_CheckMeleeRange(mo)) ?
-		mo->state->misc2 ? S_StartSound(mo, mo->state->misc2) : (void)0,
+	mo->target && (A_FaceTarget(mo), P_CheckMeleeRange(mo)) ? mo->state->misc2 ? S_Sound(mo, CHAN_BODY, SoundMap[mo->state->misc2], 1, ATTN_NORM)
+	              : (void)0,
```

Comment:
Can this be turned into a series of `if` statements or somesuch?  We don't need to repeat the mistakes of BOOM/MBF. 😄